controllers/base/namespace_controller.go

```diff
@@ -194,16 +204,41 @@ func (r *NamespaceReconciler) checkSkippedNamespace(ctx context.Context, namespa
 }
 
 // InstallStreamlitTool to install the required resource of streamlit
-func (r *NamespaceReconciler) InstallStreamlitTool(ctx context.Context, logger logr.Logger, instance *v1.Namespace) error {
+func (r *NamespaceReconciler) InstallStreamlitTool(ctx context.Context, logger logr.Logger, instance *corev1.Namespace) error {
 	logger.Info("Installing streamlit tool...")
 	stDeployer := streamlit.NewStreamlitDeployer(ctx, r.Client, instance)
 
 	return stDeployer.Install()
 }
 
 // UninstallStreamlitTool to remove resources of streamlit
-func (r *NamespaceReconciler) UninstallStreamlitTool(ctx context.Context, logger logr.Logger, instance *v1.Namespace) error {
+func (r *NamespaceReconciler) UninstallStreamlitTool(ctx context.Context, logger logr.Logger, instance *corev1.Namespace) error {
 	stDeployer := streamlit.NewStreamlitDeployer(ctx, r.Client, instance)
 
 	return stDeployer.Uninstall()
 }
+
+func (r *NamespaceReconciler) ensureRagRBAC(ctx context.Context, logger logr.Logger, instance *corev1.Namespace) error {
+	saName := env.GetString(evaluation.RAGServiceAccountEnv, evaluation.RAGJobServiceAccount)
+	sa := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName,
+			Namespace: instance.Name,
+		},
+	}
+	if err := r.Client.Create(ctx, &sa); err != nil && !errors.IsAlreadyExists(err) {
+		logger.Error(err, "failed to create serviceaccount", "ServiceAccount", evaluation.RAGJobServiceAccount, "Namespace", instance.Name)
+		return err
+	}
+	clusterRoleBindingName := env.GetString(evaluation.RAGClusterRoleBindingEnv, evaluation.RAGJobClusterRoleBinding)
+	crb := v1.ClusterRoleBinding{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: clusterRoleBindingName}, &crb); err != nil {
+		return err
+	}
+	crb.Subjects = append(crb.Subjects, v1.Subject{
```

Comment:
seems we just cover the add of subject when create NS, but didn't cover deletion of ns, and remove subject of deleted NS.